pkg/terminal/command.go

```diff
@@ -1726,10 +1727,78 @@ func toggle(t *Term, ctx callContext, args string) error {
 }
 
 func breakpoints(t *Term, ctx callContext, args string) error {
-	breakPoints, err := t.client.ListBreakpoints(args == "-a")
+	// Parse arguments
+	var showAll bool
+	var saveFile string
+
+	if args != "" {
+		argv := strings.Fields(args)
+	argsLoop:
+		for i, arg := range argv {
+			switch arg {
+			case "-a":
+				showAll = true
+			case "-save":
+				if i+1 >= len(argv) {
+					return errors.New("missing filename after -save flag")
+				}
+				saveFile = argv[i+1]
+				break argsLoop // Exit loop since we found -save and its argument
+			}
+		}
+	}
+
+	breakPoints, err := t.client.ListBreakpoints(showAll)
 	if err != nil {
 		return err
 	}
+
+	// If -save flag is provided, save breakpoints to file
+	if saveFile != "" {
+		file, err := os.OpenFile(saveFile, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0644)
+		if err != nil {
+			if os.IsExist(err) {
+				return fmt.Errorf("file '%s' already exists and cannot be overwritten", saveFile)
+			}
+			return fmt.Errorf("failed to open file '%s': %w", saveFile, err)
+		}
+		defer file.Close()
+		w := bufio.NewWriter(file)
+		defer w.Flush()
+
+		for _, bp := range breakPoints {
+			// We don't need to store these breakpoints
+			if bp.ID < 0 {
+				continue
+			}
+			var err error
+			if bp.Tracepoint {
```

Comment:
Watchpoints aren't included in this output, but if we're including tracepoints I think they should be included as well.

Also does this code work for hit conditions? E.g. `api.Breakpoint.{HitCond,HitCondPerG}`.

Reply:
I don't think it's possible to save watchpoints, the address is going to change on restart and the expression used to set them is probably not going to be evaluatable at startup. The exception to this is watchpoints set on global variables, but it's pretty complicated to which ones are those so IMO it isn't worth it.

Reply:
I was not properly saving hit conditions. I added them and also a condition to skip Watchpoints.